.github/workflows/validate-pr.yml

```diff
@@ -2,299 +2,146 @@ name: Validate PR
-
-    - name: Validate
-      run: sbt "^validate"
-  
-  scripted-universal:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    needs: [ validate ]
-    steps:
-    - uses: actions/checkout@v2
-    - name: Loading ivy cache
-      uses: actions/cache@v1
-      with:
-        path: ~/.ivy2/cache
-        key: ${{ runner.os }}-ivy-${{ hashFiles('**/*.sbt') }}
-        restore-keys: |
-          ${{ runner.os }}-ivy-
-    - name: Set up Azul JDK 1.8 and SBT
-      uses: olafurpg/setup-scala@v10
-      with:
-        java-version: zulu@1.8
-    - name: Validate
-      run: sbt "^validateUniversal"
-        
-  scripted-jar:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    needs: [ validate ]
+      matrix:
+        include:
+          - os: ubuntu-latest
+          - os: macos-latest
+    runs-on: '${{ matrix.os }}'
     steps:
-    - uses: actions/checkout@v2
-    - name: Loading ivy cache
-      uses: actions/cache@v1
-      with:
-        path: ~/.ivy2/cache
-        key: ${{ runner.os }}-ivy-${{ hashFiles('**/*.sbt') }}
-        restore-keys: |
-          ${{ runner.os }}-ivy-
-    - name: Set up Azul JDK 1.8 and SBT
-      uses: olafurpg/setup-scala@v10
-      with:
-        java-version: zulu@1.8
-    - name: Validate
-      run: sbt "^validateJar"       
-
-  scripted-bash:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    needs: [ validate ]
-    steps:
-    - uses: actions/checkout@v2
-    - name: Loading ivy cache
-      uses: actions/cache@v1
-      with:
-        path: ~/.ivy2/cache
-        key: ${{ runner.os }}-ivy-${{ hashFiles('**/*.sbt') }}
-        restore-keys: |
-          ${{ runner.os }}-ivy-
-    - name: Set up Azul JDK 1.8 and SBT
-      uses: olafurpg/setup-scala@v10
-      with:
-        java-version: zulu@1.8
-    - name: Validate
-      run: sbt "^validateBash"
-
-  scripted-ash:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    needs: [ validate ]
-    steps:
-    - uses: actions/checkout@v2
-    - name: Loading ivy cache
-      uses: actions/cache@v1
-      with:
-        path: ~/.ivy2/cache
-        key: ${{ runner.os }}-ivy-${{ hashFiles('**/*.sbt') }}
-        restore-keys: |
-          ${{ runner.os }}-ivy-
-    - name: Set up Azul JDK 1.8 and SBT
-      uses: olafurpg/setup-scala@v10
-      with:
-        java-version: zulu@1.8
-    - name: Validate
-      run: sbt "^validateAsh"
-
-  scripted-rpm:
-    runs-on: ubuntu-latest
+      - uses: actions/checkout@v4
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: zulu
+          java-version: 8
+          cache: sbt
+      - uses: sbt/setup-sbt@v1
+      - name: Validate
+        run: sbt -v "+validate"
+
+  scripted:
     strategy:
       fail-fast: false
-    needs: [ validate ]
+      matrix:
+        include:
+          - os: ubuntu-latest
+            java: 8
+            distribution: zulu
+            jobtype: 1
```

Comment:
I do like the refactoring to the matrix definition. 

What about putting the sbt command argument here directly instead of the if else stuff below? Would this work?

Reply:
Good idea. 8213f48920feb81847f50071fe54b02df6d901a7